evalme/image/object_detection.py

```diff
@@ -441,70 +441,89 @@ def distance(self, pred, local_range=1, label_weights=None, per_label=False):
+        """
+        self.algorithm = algorithm
+        # creating vars for results
         if per_label:
             results = defaultdict(float)
             num_results = defaultdict(int)
         else:
-            results = dict()
-
+            results = defaultdict(int)
+        # getting group ids from results
         gt_ids = self._get_ids_from_results()
         pred_ids = pred._get_ids_from_results()
-
+        # group result by group id and compare
         for id_gt in gt_ids:
+            # get ground truth results and types
             gt_results = self._get_results_by_id(id_gt)
-            gt_types = self._get_types_from_results(gt_results)
+            gt_types = gt_results.keys()
             for id_pred in pred_ids:
+                # get prediction results and types from current id_pred
                 pred_results = pred._get_results_by_id(id_pred)
-                pred_types = self._get_types_from_results(pred_results)
-                if 'rectangle' in pred_types and 'rectangle' in gt_types:
-                    gt_results_rectangle = [item for item in gt_results if item['type'] == 'rectangle']
-                    pred_results_rectangle = [item for item in pred_results if item['type'] == 'rectangle']
-                    score = self._get_max_iou_rectangles(gt_results_rectangle, pred_results_rectangle, threshold)
-                    if score < threshold:
-                        results[id_gt] = 0
-                    else:
-                        gt_results_labels = [item['value']['labels'] for item in gt_results if item['type'] == 'labels']
-                        pred_results_labels = [item['value']['labels'] for item in pred_results if item['type'] == 'labels']
-                        if gt_results_labels == pred_results_labels:
-                            gt_results_text = TextAreaEvalItem([item for item in gt_results if item['type'] != 'labels' and item['type'] != 'rectangle'])
-                            pred_results_text = TextAreaEvalItem([item for item in pred_results if item['type'] != 'labels' and item['type'] != 'rectangle'])
-                            res = gt_results_text.match(item=pred_results_text, algorithm=algorithm)
+                pred_types = pred_results.keys()
+                # Tag for region selection [check OCR_SHAPES list]
+                for rec_type in OCREvalItem.OCR_SHAPES:
+                    # check if both groups have region selection tags
+                    if rec_type in pred_types and rec_type in gt_types:
+                        # check labels and compare labels
+                        gt_results_labels = gt_results.get('labels', [])
+                        pred_results_labels = pred_results.get('labels', [])
+                        if len(gt_results_labels) > 1 \
+                                or len(pred_results_labels) > 1 \
+                                or gt_results_labels[0]['value']['labels'] != pred_results_labels[0]['value']['labels']:
```

Comment:
What if we have more control tags than one in the labeling config? E.g. audio + ocr.

Reply:
Works good, added test for this case